pkg/hubble/parser/options/options.go

```diff
@@ -27,14 +41,27 @@ func CacheSize(size int) Option {
 }
 
 // Redact configures which data Hubble will redact.
-func Redact(logger logrus.FieldLogger, httpQuery bool, kafkaApiKey bool) Option {
+func Redact(logger logrus.FieldLogger, httpQuery, kafkaApiKey bool, allowHeaders, denyHeaders []string) Option {
 	return func(opt *Options) {
-		opt.RedactHTTPQuery = httpQuery
-		opt.RedactKafkaAPIKey = kafkaApiKey
+		opt.HubbleRedactSettings.Enabled = true
+		opt.HubbleRedactSettings.RedactHTTPQuery = httpQuery
+		opt.HubbleRedactSettings.RedactKafkaAPIKey = kafkaApiKey
+		opt.HubbleRedactSettings.RedactHttpHeaders = HttpHeadersList{
+			Allow: headerSliceToMap(allowHeaders),
+			Deny:  headerSliceToMap(denyHeaders),
+		}
 		if logger != nil {
 			logger.WithField(
 				"options",
 				fmt.Sprintf("%+v", opt)).Info("configured Hubble with redact options")
 		}
 	}
 }
+
+func headerSliceToMap(headerList []string) map[string]struct{} {
+	headerMap := map[string]struct{}{}
```

Comment:
Given that we know the size of the map in advance, we can preallocate memory for the map to avoid multiple re-allocation when adding new keypair values.
```suggestion
	headerMap := make(map[string]struct{}, len(headerList))
```